core/services/ocr2/plugins/ocr2keeper/evm21/services.go

```diff
@@ -69,7 +69,7 @@ func New(addr common.Address, client evm.Chain, mc *models.MercuryCredentials, k
 	scanner := upkeepstate.NewPerformedEventsScanner(lggr, client.LogPoller(), addr)
 	services.upkeepState = upkeepstate.NewUpkeepStateStore(lggr, scanner)
 
-	logProvider, logRecoverer := logprovider.New(lggr, client.LogPoller(), client.Client(), utilsABI, services.upkeepState)
+	logProvider, logRecoverer := logprovider.New(lggr, client.LogPoller(), client.Client(), utilsABI, services.upkeepState, client.Config().EVM().FinalityDepth())
```

Comment:
🙌

Reply:
I checked finality depth for polygon is defined as 500 https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/config/toml/defaults/Polygon_Mainnet.toml#L4 so great we added this!